datastore/src/main/java/org/jboss/pnc/datastore/repositories/BuildRecordRepositoryImpl.java

```diff
@@ -133,6 +134,14 @@ public BuildRecord getLatestSuccessfulBuildRecord(Integer configurationId, boole
         return getLatestSuccessfulBuildRecord(buildRecords, temporaryBuild);
     }
 
+    @Override
+    public BuildRecord getAnyLatestSuccessfulBuildRecordWithBuildConfig(
+            Integer configurationId,
+            boolean temporaryBuild) {
+        List<BuildRecord> buildRecords = queryWithBuildConfigurationId(configurationId);
+        return getAnyLatestSuccessfulBuildRecord(buildRecords, temporaryBuild);
```

Comment:
Maybe we can use simmilar method to `getAnyLatestSuccessfulBuildRecordWithRevision` but with `withBuildConfigurationId` instead of `withBuildConfigurationIdRev`. Then we don't need `getAnyLatestSuccessfulBuildRecord` and we are getting less data from DB.

I guess before the logic was too complex for easy query like this.
```suggestion
        PageInfo pageInfo = new DefaultPageInfo(0, 1);
        SortInfo sortInfo = new DefaultSortInfo(SortInfo.SortingDirection.DESC, BuildRecord_.submitTime.getName());

        List<BuildRecord> buildRecords = queryWithPredicates(
                pageInfo,
                sortInfo,
                withBuildConfigurationId(configurationId),
                withSuccess(),
                includeTemporary(temporaryBuild));

        if (buildRecords.size() == 0) {
            return null;
        } else {
            return buildRecords.get(0);
        }
```

Reply:
I like it much better yes, thanks for that, I overlooked the possible optimization although I never liked the way it retrieved all the build records before. Thanks!